implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py

```diff
@@ -0,0 +1,147 @@
+        return True
+    return args["request"]["responseType"] == HttpResponseType.BINARY
+
+
+class HttpPlugin(PluginModule[None]):
+    def __init__(self):
+        super().__init__(None)
+        self.client = AsyncClient()
+
+    async def get(self, args: ArgsGet, invoker: Invoker) -> Result[HttpResponse]:
+        res: Response
+        if args.get("request") is None:
+            res = await self.client.get(args["url"])
+        elif args["request"] is not None:
+            res = await self.client.get(
+                args["url"],
+                params=args["request"]["urlParams"],
+                headers=args["request"]["headers"],
+                timeout=cast(float, args["request"]["timeout"]),
+            )
+        else:
+            res = await self.client.get(args["url"])
+
+        if isResponseBinary(args):
+            return Ok(
+                HttpResponse(
+                    status=res.status_code,
+                    statusText=res.reason_phrase,
+                    headers=GenericMap(dict(res.headers)),
+                    body=base64.b64encode(res.content).decode(),
+                )
+            )
+
+        return Ok(
+            HttpResponse(
+                status=res.status_code,
+                statusText=res.reason_phrase,
+                headers=GenericMap(dict(res.headers)),
+                body=res.text,
+            )
+        )
+
+    async def post(self, args: ArgsPost, invoker: Invoker) -> Result[HttpResponse]:
+        res: Response
+        if args.get("request") is None:
+            res = await self.client.post(args["url"])
+        elif args["request"] is not None:
+            content = (
+                args["request"]["body"].encode()
+                if args["request"]["body"] is not None
+                else None
+            )
+            res = await self.client.post(
+                args["url"],
+                content=content,
+                params=args["request"]["urlParams"],
+                headers=args["request"]["headers"],
+                timeout=cast(float, args["request"]["timeout"]),
+            )
+        else:
+            res = await self.client.post(args["url"])
+
+        if args["request"] is not None and args["request"]["responseType"] == HttpResponseType.BINARY:
+            return Ok(
+                HttpResponse(
+                    status=res.status_code,
+                    statusText=res.reason_phrase,
+                    headers=GenericMap(dict(res.headers)),
+                    body=base64.b64encode(res.content).decode(),
+                )
+            )
+
+        return Ok(
+            HttpResponse(
+                status=res.status_code,
+                statusText=res.reason_phrase,
+                headers=GenericMap(dict(res.headers)),
+                body=res.text,
+            )
+        )
+
+
+def http_plugin():
+    manifest_path = Path(__file__).parent.joinpath("manifest.json")
```

Comment:
Ideally we don't want to load this via the file-system, in cases where the python program doesn't allow access.

Reply:
There will be parse wrap.info.py in the future like js when we add codegen for python plugin

Reply:
Okay sounds good